cpu/native/socket_zep/socket_zep.c

```diff
@@ -0,0 +1,424 @@
+    uint16_t *v = value;
+
+    assert((dev != NULL) && (value != NULL));
+    switch (opt) {
+        case NETOPT_MAX_PACKET_SIZE:
+            if (max_len != sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *v = SOCKET_ZEP_FRAME_PAYLOAD_LEN;
+            return sizeof(uint16_t);
+        default:
+            return netdev2_ieee802154_get(&dev->netdev, opt, value, max_len);
+    }
+}
+
+static int _set(netdev2_t *netdev, netopt_t opt, void *value, size_t value_len)
+{
+    assert(netdev != NULL);
+    return netdev2_ieee802154_set((netdev2_ieee802154_t *)netdev, opt,
+                                  value, value_len);
+}
+
+
+void socket_zep_setup(socket_zep_t *dev, const socket_zep_params_t *params)
+{
+    static const struct addrinfo hints = { .ai_family = AF_UNSPEC,
+                                           .ai_socktype = SOCK_DGRAM };
+    struct addrinfo *ai = NULL, *remote;
+    int res;
+
+    assert((params->local_addr != NULL) && (params->local_port != NULL) &&
+           (params->remote_addr != NULL) && (params->remote_port != NULL));
+    memset(dev, 0, sizeof(socket_zep_t));
+    dev->netdev.netdev.driver = &socket_zep_driver;
+    /* bind and connect socket */
+    if ((res = real_getaddrinfo(params->local_addr, params->local_port, &hints,
+                                &ai)) < 0) {
+        fprintf(stderr, "ZEP: unable to get local address: %s\n",
+                gai_strerror(res));
+        return;
+    }
+    for (struct addrinfo *local = ai; local != NULL; local = local->ai_next) {
+        if ((res = real_socket(local->ai_family, local->ai_socktype,
+                               local->ai_protocol)) < 0) {
+            continue;
+        }
+        if (real_bind(res, local->ai_addr, local->ai_addrlen) == 0) {
+            break;  /* successfully bound */
+        }
+    }
+    real_freeaddrinfo(ai);
+    if (res < 0) {
+        perror("ZEP: Unable to bind socket");
+        return;
+    }
+    dev->sock_fd = res;
+    ai = NULL;
+    if ((res = real_getaddrinfo(params->remote_addr, params->remote_port, &hints,
+                                &ai)) < 0) {
+        fprintf(stderr, "ZEP: unable to get remote address: %s\n",
+                gai_strerror(res));
+        return;
+    }
+    for (remote = ai; remote != NULL; remote = remote->ai_next) {
+        if (real_connect(dev->sock_fd, remote->ai_addr, remote->ai_addrlen) == 0) {
+            break;  /* successfully connected */
+        }
+    }
+    if (remote == NULL) {
+        perror("ZEP: Unable to connect socket");
+        return;
+    }
+    real_freeaddrinfo(ai);
+
+    /* generate hardware address from local address */
+    uint8_t ss_array[sizeof(struct sockaddr_storage)] = { 0 };
+    socklen_t ss_len = sizeof(struct sockaddr_storage);
+
+    if (getpeername(dev->sock_fd, (struct sockaddr *)&ss_array, &ss_len) < 0) {
+        perror("ZEP: Unable to retrieve remote address");
+        return;
```

Comment:
I think that's a remnant from when I played around with unbound/unconnected sockets. Will do.